### PR TITLE
KFSPTS-17515 add KualiCo checkstyle rules and some Cornell specific c…

### PR DESCRIPTION
**Pull request by db483**
…hecks and apply to cam module only

### CODE REVIEW

File: src/main/config/CU_Checkstyle.xml [hunk 1]
```diff
@@ -5,421 +5,205 @@
...
-		
-		<module name="MemberName">
-			<property name="severity" value="info"/>
-		</module>
-
-		<module name="MethodLength">
-			<property name="severity" value="info"/>
-			<property name="max" value="150"/>
-			<property name="countEmpty" value="false"/>
-		</module>
```
Comment (cah292) [inline src/main/config/CU_Checkstyle.xml]: Our Confluence documentation mentions that for this rule, we should "keep it commented." If doing so results in our configuration not being clean enough, then can a separate user story be created to potentially revisit this rule?

File: src/main/config/CU_Checkstyle.xml [hunk 2]
```diff
@@ -5,421 +5,205 @@
...
+                        org.aspectj,
+                        org.hsqldb,
+                        org.jdom,
+                        org.kuali.rice.location,
+                        org.objectweb.asm.signature,
+                        org.w3c.dom.html,
+                        org.apache.log4j"
+            />
+            <property name="illegalClasses"
+                      value="org.kuali.rice.kns.service.TransactionalDocumentDictionaryService"
```
[the remark this answers is not in the document]
Comment (db483) [reply src/main/config/CU_Checkstyle.xml]: I have updated it to be based on the 8/15 release. We should update the checkstyle with each upgrade.

File: src/main/config/CU_Checkstyle.xml [hunk 3]
```diff
@@ -5,421 +5,205 @@
...
+        <module name="TrailingComment"/>
+        <module name="TypecastParenPad"/>
+        <module name="UnnecessaryParentheses"/>
+        <module name="UnusedImports"/>
+        <module name="UpperEll"/>
+        <module name="WhitespaceAfter"/>
+        <module name="WhitespaceAround"/>
+        <!-- Cornell specific checks --> 
+        <module name="AnnotationUseStyle">
+      		<property name="severity" value="error"/>
```
[the remark this answers is not in the document]
Comment (db483) [reply src/main/config/CU_Checkstyle.xml]: tabs removed

File: src/main/config/CU_Checkstyle.xml [hunk 4]
```diff
@@ -5,421 +5,205 @@
...
+      		<property name="elementStyle" value="ignore"/>
+      		<property name="closingParens" value="ignore"/>
+      		<!-- "never" because Sun's compiler cannot handle a trailing comma and will not compile (who broke the build?) -->
+      		<property name="trailingArrayComma" value="never"/>
 		</module>
-		
-		<module name="IllegalType">
+	    <module name="IllegalType">
 			<property name="severity" value="warning"/>
```
[the remark this answers is not in the document]
Comment (db483) [reply src/main/config/CU_Checkstyle.xml]: updated

File: src/main/config/CU_Checkstyle.xml [hunk 5]
```diff
@@ -5,421 +5,205 @@
...
             <property name="message" value="Use of System.out.* method"/>
             <property name="ignoreComments" value="true"/>
         </module>
-        
-        <module name="RegexpSinglelineJava">
-        	<property name="severity" value="warning"/>
-            <property name="format" value="System\.err*$"/>
-            <property name="message" value="Use of System.err.* method"/>
-            <property name="ignoreComments" value="true"/>
-        </module>
```
[the remark this answers is not in the document]
Comment (db483) [reply src/main/config/CU_Checkstyle.xml]: I have added the rule back

File: src/main/config/checkstyle-suppressions.xml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+-->
+<!DOCTYPE suppressions PUBLIC
+        "-//Puppy Crawl//DTD Suppressions 1.1//EN"
+        "http://checkstyle.sourceforge.net/dtds/suppressions_1_1.dtd">
+<suppressions>
+    <suppress checks=".*" files="src[\\/]main[\\/]java[\\/]com[\\/]rsmart[\\/]kuali" />
+    <suppress checks=".*" files="src[\\/]main[\\/]java[\\/]org[\\/]kuali[\\/]kfs" />
+    <suppress checks=".*" files="src[\\/]main[\\/]java[\\/]util" />
+    <suppress checks=".*" files="src[\\/]main[\\/]java[\\/]cynergy" />
+    <suppress checks=".*" files="src[\\/]main[\\/]java[\\/]util" />
```
[the remark this answers is not in the document]
Comment (db483) [reply src/main/config/checkstyle-suppressions.xml]: removed duplicate

File: src/main/config/checkstyle-suppressions.xml [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    <suppress checks=".*" files="src[\\/]main[\\/]java[\\/]edu[\\/]cornell[\\/]kfs[\\/]tax" />
+    <suppress checks=".*" files="src[\\/]main[\\/]java[\\/]edu[\\/]cornell[\\/]kfs[\\/]vnd" />
+    <suppress checks=".*" files="src[\\/]main[\\/]java[\\/]edu[\\/]cornell[\\/]kfs[\\/]web" />
+    <suppress checks=".*" files="src[\\/]main[\\/]java[\\/]edu[\\/]cornell[\\/]kfs[\\/]module[\\/]ar" />
+    <suppress checks=".*" files="src[\\/]main[\\/]java[\\/]edu[\\/]cornell[\\/]kfs[\\/]module[\\/]cg" />
+    <suppress checks=".*" files="src[\\/]main[\\/]java[\\/]edu[\\/]cornell[\\/]kfs[\\/]module[\\/]ezra" />
+    <suppress checks=".*" files="src[\\/]main[\\/]java[\\/]edu[\\/]cornell[\\/]kfs[\\/]module[\\/]ld" />
+    <suppress checks=".*" files="src[\\/]main[\\/]java[\\/]edu[\\/]cornell[\\/]kfs[\\/]module[\\/]purap" />
+    <suppress checks=".*" files="src[\\/]main[\\/]java[\\/]edu[\\/]cornell[\\/]kfs[\\/]module[\\/]receiptProcessing" />
+    <suppress checks=".*" files="src[\\/]main[\\/]java[\\/]edu[\\/]cornell[\\/]kfs[\\/]module[\\/]ar" />
```
[the remark this answers is not in the document]
Comment (db483) [reply src/main/config/checkstyle-suppressions.xml]: removed duplicate

**Comment by db483** [comment]
[1] conflicts have been resolved
[2] https://jira.cornell.edu/browse/KFSPTS-17980 has been created

File: src/main/config/CU_Checkstyle.xml (hunk 5 above)
[the remark this answers is not in the document]
Comment (db483) [reply src/main/config/CU_Checkstyle.xml]: added missing rule

File: src/main/config/checkstyle-suppressions.xml (hunk 6 above)
[the remark this answers is not in the document]
Comment (db483) [reply src/main/config/checkstyle-suppressions.xml]: duplicate removed

File: src/main/config/checkstyle-suppressions.xml [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+-->
+<!DOCTYPE suppressions PUBLIC
+        "-//Puppy Crawl//DTD Suppressions 1.1//EN"
+        "http://checkstyle.sourceforge.net/dtds/suppressions_1_1.dtd">
+<suppressions>
+    <suppress checks=".*" files="src[\\/]main[\\/]java[\\/]com[\\/]rsmart[\\/]kuali" />
```
[the remark this answers is not in the document]
Comment (db483) [reply src/main/config/checkstyle-suppressions.xml]: I think this works as expected. Could you also check locally? Thank you!

File: src/main/config/CU_Checkstyle.xml (hunk 1 above)
Comment (db483) [reply src/main/config/CU_Checkstyle.xml]: I have created https://jira.cornell.edu/browse/KFSPTS-17981

File: src/main/config/checkstyle-suppressions.xml (hunk 8 above)
[the remark this answers is not in the document]
Comment (cah292) [reply src/main/config/checkstyle-suppressions.xml]: I tested out the changes from this feature branch locally, and the exclusions seem to be working as expected after I forced Eclipse to do a full Checkstyle check on the cu-kfs project again. Thanks!

**Pull request merged**